enclave-runtime/src/rpc/author/mod.rs

```diff
@@ -170,6 +184,13 @@ where
 			Ok(op) => op,
 			Err(_) => return Box::pin(ready(Err(ClientError::BadFormat.into()))),
 		};
+
+		// apply top filter - return error if this specific type of trusted operation
+		// is not allowed by the filter
+		if let None = self.top_filter.filter(&stf_operation) {
+			return Box::pin(ready(Err(ClientError::UnsupportedOperation.into())))
+		}
+
```

Comment:
When submitting the trusted operation, we check if it is filtered out or not. If it is filtered out (i.e. not allowed), we return an error.